### PR TITLE
docs: known issue for listener using `proxy_protocol _behavior`

### PR DESCRIPTION
**Pull request by peteski22**
### Description

Documents a known issue which impacts various versions of Vault.

### TODO only if you're a HashiCorp employee

- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [x] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] ~**RFC:** If this change has an associated RFC, please link it in the description.~
- [ ] ~**ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.~

### CODE REVIEW

File: website/content/docs/release-notes/1.17.0.mdx [hunk 1]
```diff
@@ -13,14 +13,15 @@ description: |-
...
+| Beta feature deprecated (1.17) | [Request limiter deprecated](/vault/docs/upgrading/upgrade-to-1.17.x#request-limiter)                                                                               |
+| Known issue (1.17.0+)          | [PKI OCSP GET requests can return HTTP redirect responses](/vault/docs/upgrading/upgrade-to-1.17.x#pki-ocsp)                                                        |
+| Known issue (1.17.0)           | [Vault Agent and Vault Proxy consume excessive amounts of CPU](/vault/docs/upgrading/upgrade-to-1.17.x#agent-proxy-cpu-1-17)                                        |
+| Known issue (1.15.8+)          | [Autopilot upgrade for Vault Enterprise fails](/vault/docs/upgrading/upgrade-to-1.15.x#autopilot)                                                                   |
+| Known issue (1.17.1)           | [Listener stops listening on untrusted upstream connection with particular config settings](/vault/docs/upgrading/upgrade-to-1.17.x#listener-proxy-protocol-config) |
```
Comment (VioletHynes) [inline website/content/docs/release-notes/1.17.0.mdx:24]:
```suggestion
| Known issue (1.17.0 - 1.17.1)           | [Listener stops listening on untrusted upstream connection with particular config settings](/vault/docs/upgrading/upgrade-to-1.17.x#listener-proxy-protocol-config) |
```

Comment (VioletHynes) [reply website/content/docs/release-notes/1.17.0.mdx:24]: This is present in 1.17.0 also, right? We should put both versions

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<a id="listener-proxy-protocol-config" />
+
+### Potential DoS when using the `deny_unauthorized` proxy protocol behavior for a TCP listener
+
+#### Affected versions
+
+##### Community Edition (CE)
+
+- 1.10.x - 1.15.x
+- 1.17.1
```
Comment (VioletHynes) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:10]: Should this be 1.17.0 and 1.17.1?

File: website/content/docs/release-notes/1.17.0.mdx (hunk 1 above)
Comment (peteski22) [reply website/content/docs/release-notes/1.17.0.mdx:24]: No. It's only present in `1.17.1` in our `1.17.x` releases.

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx (hunk 2 above)
Comment (peteski22) [reply website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:10]: No only `1.17.1`

**Review by VioletHynes — APPROVED** [review]
Looks good to me! Great detail -- and sorry for the confusion on the versions.

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+- 1.10.x+ent - 1.15.11+ent
+- 1.16.5+ent
+- 1.17.1+ent
+
+#### Issue
+
+When a Vault TCP listener is configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
+and receives a request from an untrusted upstream connection, Vault will no longer
+respond to requests via that listener.
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:22]:
```suggestion
Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior) process, but do not respond to, requests from untrusted upstream connections,
```
Style correction: write in active voice

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+This could lead to a situation where the Vault API could be vulnerable to a denial-of-service
+(DoS) attack, if requests are intentionally made by untrusted upstream connections
+to the Vault API, causing it to become unavailable on that node.
+
+This issue is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:28]:
```suggestion
The fact that the listener takes the time to process the request before ignoring it, leaves the listener vulnerable to denial-of-service (DoS) attacks by requests intentionally sent from untrusted upstream connections.

We recommend upgrading to Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent or later to address the vulnerability.
```

Style correction: write in active voice

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx (hunk 4 above)
Comment (peteski22) [reply website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:28]: > The fact that the listener takes the time to process the request before ignoring it, leaves the listener 

The vulnerability is such that you can prevent Vault from serving its API entirely - therefore no further requests can be successful to that Vault node. 

It's not that the listener takes a long time to process anything, it doesn't it, it's quick to process but closes when we stop serving HTTP in response to the way the library returns the error from the listener's `Accept` method.

Comment (peteski22) [reply website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:28]: > We recommend upgrading to Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent or later to address the vulnerability.

These versions aren't out yet, so I was just following the pattern we've used in previous known issues to call out the versions that (will) fix it.

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- 1.16.5+ent
+- 1.17.1+ent
+
+#### Issue
+
+Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
+which receive a request from an untrusted upstream connection will close, causing
+Vault to no longer respond to requests via that listener.
+
+This leaves the Vault API vulnerable to a denial-of-service (DoS) attack by requests
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:24]: What is "this"? The closed listener or the fact that Vault can't receive requests through it?

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+- 1.10.x+ent - 1.15.11+ent
+- 1.16.5+ent
+- 1.17.1+ent
+
+#### Issue
+
+Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
+which receive a request from an untrusted upstream connection will close, causing
+Vault to no longer respond to requests via that listener.
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:22]:
```suggestion
Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
close if they receive a request from an untrusted upstream connection. As a result,
Vault no longer responds to *any* request received through that listener.
```

Style correction: write in active voice

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+This leaves the Vault API vulnerable to a denial-of-service (DoS) attack by requests
+which are intentionally made by untrusted upstream connections, causing it to become
+unavailable on that node.
+
+This issue is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:28]: What is "this"? The fact that the listener closes or that it leaves Vault susceptible to DoS attacks?

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx (hunk 7 above)
Comment (peteski22) [reply website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:28]: Both,  by stopping the listener from closing we stop the ability for the DoS to be exploited in this way.

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Issue
+
+Vault TCP listeners configured with the `deny_unauthorized` [proxy_protocol_behavior](/vault/docs/configuration/listener/tcp#proxy_protocol_behavior)
+close if they receive a request from an untrusted upstream connection. As a result,
+Vault no longer responds to *any* request received through that listener.
+
+Being able to close listeners in this way leaves the Vault API vulnerable to a
+denial-of-service (DoS) attack by requests which are intentionally made by untrusted
+upstream connections, causing it to become unavailable on that node.
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:26]:
```suggestion
Being able to force-close listeners with intentionally untrusted connections
leaves the Vault API vulnerable to a denial-of-service (DoS) attacks, which
could cause the API to become unavailable on that node.
```

File: website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+This issue is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.
```
Comment (schavis) [inline website/content/partials/known-issues/config_listener_proxy_protocol_behavior_issue.mdx:28]:
```suggestion
The vulnerability is addressed in Vault 1.15.12+ent, 1.16.6+ent, 1.17.2, 1.17.2+ent and later.
```

**Review by schavis — APPROVED** [review]
Approving to unblock publication. Please be sure to apply the suggestions before merging.

**Pull request merged**